bundles/org.openhab.binding.energenie/src/main/java/org/openhab/binding/energenie/internal/handler/EnergeniePWMHandler.java

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.energenie.internal.handler;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.energenie.internal.EnergeniePWMStateEnum;
+import org.openhab.binding.energenie.internal.config.EnergenieConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EnergeniePWMHandler} is responsible for reading states and update PWM channels.
+ *
+ * @author Hans-Jörg Merk - Initial contribution
+ */
+
+@NonNullByDefault
+public class EnergeniePWMHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(EnergeniePWMHandler.class);
+
+    private static final int HTTP_TIMEOUT_MILLISECONDS = 6000;
+
+    private @Nullable EnergenieConfiguration config;
+
+    private String host = "";
+    private String password = "";
+    private int refreshInterval;
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    public EnergeniePWMHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            try {
+                scheduler.execute(this::getState);
+            } catch (Exception e) {
+                logger.debug("Exception during poll", e);
+            }
```

Comment:
You can remove this try/catch. There aren't any exceptions that you should be catching here.

Reply:
Done